src/Cogworks.FindAndReplace/Models/Dtos/DatabaseDtos/PropertyDataDto.cs

```diff
@@ -0,0 +1,96 @@
+﻿using NPoco;
+using Umbraco.Core.Persistence.DatabaseAnnotations;
+
+namespace Cogworks.FindAndReplace.Models.Dtos.DatabaseDtos
+{
+    [TableName(TableName)]
+    [PrimaryKey("id")]
+    [ExplicitColumns]
+    public class PropertyDataDto
+    {
+        public const string TableName = Constants.DatabaseSchema.Tables.PropertyData;
+        public const int VarcharLength = 512;
+
+        [Column("id")]
+        [PrimaryKeyColumn]
+        public int Id { get; set; }
+
+        [Column("versionId")]
+        [ForeignKey(typeof(ContentVersionDto))]
+        [Index(IndexTypes.UniqueNonClustered, Name = "IX_" + TableName + "_VersionId", ForColumns = "versionId,propertyTypeId,languageId,segment")]
+        public int VersionId { get; set; }
+
+        [Column("propertyTypeId")]
+        [ForeignKey(typeof(PropertyTypeDto))]
+        [Index(IndexTypes.NonClustered, Name = "IX_" + TableName + "_PropertyTypeId")]
+        public int PropertyTypeId { get; set; }
+
+        [Column("languageId")]
+        [ForeignKey(typeof(LanguageDto))]
+        [Index(IndexTypes.NonClustered, Name = "IX_" + TableName + "_LanguageId")]
+        [NullSetting(NullSetting = NullSettings.Null)]
+        public int? LanguageId { get; set; }
+
+
+        [Column("varcharValue")]
+        [NullSetting(NullSetting = NullSettings.Null)]
+        [Length(VarcharLength)]
+        public string VarcharValue { get; set; }
+
+        [Column("textValue")]
+        [NullSetting(NullSetting = NullSettings.Null)]
+        [SpecialDbType(SpecialDbTypes.NTEXT)]
+        public string TextValue { get; set; }
+
+        [ResultColumn]
+        [Reference(ReferenceType.OneToOne, ColumnName = "PropertyTypeId")]
+        public PropertyTypeDto PropertyTypeDto { get; set; }
+
+        [Ignore]
+        public object Value
+        {
+            get
+            {
+                if (!string.IsNullOrEmpty(VarcharValue))
```

Comment:
Should we make these "IsNullOrWhitespace" instead?

Reply:
This is from Umbraco.Core. I didn't touch it.